src/shell/shell.cpp

```diff
@@ -70,67 +70,129 @@ DOS_Shell::DOS_Shell()
 	}
 }
 
-void DOS_Shell::GetRedirection(char *line,
-                               std::string &in_file,
-                               std::string &out_file,
-                               std::string &pipe_file,
-                               bool *append)
+// This function gets redirection targets from the given shell command line and
+// returns the results as a struct. The results include the redirection targets
+// as well as the processed command line with the targets stripped off.
+//
+// Note that real MS-DOS is quite nuanced in its whitespace handling:
+// - 'echo 1>out.txt' produces a 3-byte file: '1CRLF'
+// - 'echo 1 > out.txt' produces a 4-byte file: '1 CRLF'
+// - 'echo 1 >out.txt ' produces a 5-byte file: '1  CRLF'
+// This behavior is replicated here; see the shell_redirection_tests for more
+// examples.
+//
```

Comment:
Excellent comment documenting the quirky MS-DOS behaviour @interloper98 🏅

Reply:
I was surprised by it! At least the redirection results and unit tests handle this whitespace correctly.

That said, our downstream shell and command-execution code that use the results still strips that whitespace in various places (and trying to fix that opened a bit can of worms.. so I stepped back :laughing: ), so it's a bit pedantic right now :slightly_smiling_face: .. but one more step in the right direction.